src/getting-started.md

```diff
@@ -0,0 +1,338 @@
+automatically clone and sync them (more later).
+
+[**Take a look at the "Suggested Workflows" chapter for some helpful
+advice.**][suggested]
+
+[suggested]: ./building/suggested.md
+
+### System Requirements
+
+[**See this chapter for detailed software requirements.**](./building/prerequisites.md)
+Most notably, you will need Python 2 to run `x.py`.
+
+There are no hard hardware requirements, but building the compiler is
+computationally expensive, so a beefier machine will help, and I wouldn't
+recommend trying to build on a Raspberry Pi :P
+
+- x86 and ARM are both supported (TODO: confirm)
+- Recommended >=30GB of free disk space; otherwise, you will have to keep
+  clearing incremental caches. More space is better, the compiler is a bit of a
+  hog; it's a problem we are aware of.
+- Recommended >=8GB RAM.
+- Recommended >=2 cores; more cores really helps.
+- You will need an internet connection to build; the bootstrapping process
+  involves updating git submodules and downloading a beta compiler. It doesn't
+  need to be super fast, but that can help.
+
+Building the compiler takes more than half an hour on my moderately powerful
+laptop. The first time you build the compiler, LLVM will also be built unless
+you use system LLVM (see below).
+
+Like `cargo`, the build system will use as many cores as possible. Sometimes
+this can cause you to run low on memory. You can use `-j` to adjust the number
+concurrent jobs.
+
+Also, if you don't have too much free disk space, you may want to turn off
+incremental compilation (see the "Configuring" section below). This will make
+compilation take longer, but will save a ton of space from the incremental
+caches.
+
+### Cloning
+
+You can just do a normal git clone:
+
+```shell
+git clone https://github.com/rust-lang/rust.git
+```
+
+You don't need to clone the submodules at this time.
+
+**Pro tip**: if you contribute often, you may want to look at the git worktrees
+tip in [this chapter][suggested].
+
+### Configuring the Compiler
+
+The compiler has a configuration file which contains a ton of settings. We will
+provide some recommendations here that should work for most, but [check out
+this chapter for more info][config].
+
+[config]: ./building/how-to-build-and-run.html#create-a-configtoml
+
+In the top level of the repo:
+
+```shell
+cp config.toml.example config.toml
+```
+
+Then, edit `config.toml`. You will need to search for, uncomment, and update
+the following settings:
+
+- `debug = true`: enables debug symbols and `debug!` logging, takes a bit longer to compile.
+- `incremental = true`: enables incremental compilation of the compiler itself.
+  This is turned off by default because it's technically unsound. Sometimes
+  this will cause weird crashes, but it can really speed things up.
+- `llvm-config`: enable building with system LLVM. [See this chapter][sysllvm]
+  for more info. This avoids building LLVM, which can take a while.
```

Comment:
Do you want to mention the rough timescale of how long it takes? I know that I would have liked to know in advance roughly how long it takes.

I know that it definitely varies across different computers, but mentioning a general range might be helpful. E.g., for me, I think it took around 30 minutes.

Reply:
Also mention that it doesn't have to be rebuilt that often?

Reply:
> for me, I think it took around 30 minutes.

It took me a while the first time (maybe ~15 minutes?) but incremental rebuilds are pretty fast, maybe 1-3 minutes.

Reply:
@jyn514 I'm curious what are you rebuilding and what is your hardware? In my experience, it depends on how much you need to rebuild (e.g. if you touch `librustc`, then you need to rebuild a bunch of stuff)...

Reply:
Oh sorry I meant just for LLVM. Rebuilding rustc takes much longer.